pkgs/native_assets_builder/test/build_runner/build_planner_test.dart

```diff
@@ -76,29 +76,32 @@ void main() async {
     });
   });
 
-  test('runPackageName', () async {
-    await inTempDir((tempUri) async {
-      await copyTestProjects(targetUri: tempUri);
-      final nativeAddUri = tempUri.resolve('native_add/');
+  for (final existing in [true, false]) {
```

Comment:
This could also be a split into two tests sharing logic, this seems a bit complicated at a first glance

Reply:
It's a rather common pattern we have in the code base to test variants. If we want a different pattern (with function calls instead of outer for loops) we should adopt it everywhere.